AEPMessaging/Sources/Messaging.swift

```diff
@@ -117,17 +117,31 @@ public class Messaging: NSObject, Extension {
     /// Generates and dispatches an event prompting the Edge extension to fetch in-app messages.
     /// The app surface used in the request is generated using the `bundleIdentifier` for the app.
     /// If the `bundleIdentifier` is unavailable, calling this method will do nothing.
-    private func fetchMessages() {
+    private func fetchMessages(for surfaces: [String]? = nil) {
         guard appSurface != "unknown" else {
-            Log.warning(label: MessagingConstants.LOG_TAG, "Unable to retrieve in-app messages - unable to retrieve bundle identifier.")
+            Log.warning(label: MessagingConstants.LOG_TAG, "Unable to retrieve in-app or feed messages, cannot read the bundle identifier.")
+            return
+        }
+        
+        var targetSurfaces: [String] = []
+        if let surfaces = surfaces {
+            targetSurfaces = surfaces
+                .filter { !$0.isEmpty }
+                .map { appSurface + MessagingConstants.PATH_SEPARATOR + $0 }
```

Comment:
Should we also validate here that the target surface makes a valid uri?

Reply:
ya, i think that's a good idea.